lib/sycamore/sycamore/writer.py

```diff
@@ -695,6 +692,7 @@ def json(
             resource_args: Arguments to pass to the underlying execution environment.
         """
 
-        node = JsonWriter(self.plan, path, filesystem=filesystem, **resource_args)
+        node: Node = JsonWriter(self.plan, path, filesystem=filesystem, **resource_args)
```

Comment:
ds = DocSet(self.context, node)
ds.execute()

like the other simple ones, I think this would work.

Reply:
Copying from slack for bookkeeping:

I think that would work but I'm worried about a double-execution - the internal `execute` method of the json/filewriter is basically "execute child to get ray dataset, write ray dataset, return ray dataset" and I think the "write ray dataset" forces execution, but so does the docset.execute iterrows bit.